browser/src/service/NodeAutosuggestService.tsx

```diff
@@ -1,15 +1,71 @@
 import React from "react"
 
-export class NodeAutosuggestService extends React.Component{
-    
-    add(a: number, b: number){
-        return a+b
+export class NodeAutosuggestService {
+    //constructor is empty, only necessary to access methods parseString, and addQuestionsMarks
+    constructor() {}
+
+    /* 
+    Component is passed in and a shorter name of the component is returned, 
+    Time Complexity O(n)
+    Space Complexity O(n)
+    */
+    parseString(component: string) {
```

Comment:
This isn't a generic "string parser" -- we're parsing a Dagger binding key. Consider `displayNameForKey()` or similar.

Reply:
Also move this to a helper/util file, since it doesn't relate to "NoteAutoSuggestService" and will be used in several places.